packages/flutter_tools/test/src/fake_process_manager.dart

```diff
@@ -82,39 +82,14 @@ class FakeCommand {
   /// resolves.
   final Completer<void> completer;
 
-  static bool _listEquals<T>(List<T> a, List<T> b) {
-    if (a == null) {
-      return b == null;
-    }
-    if (b == null || a.length != b.length) {
-      return false;
-    }
-    for (int index = 0; index < a.length; index += 1) {
-      if (a[index] != b[index]) {
-        return false;
-      }
-    }
-    return true;
-  }
-
-  bool _matches(List<String> command, String workingDirectory, Map<String, String> environment) {
-    if (!_listEquals(command, this.command)) {
-      return false;
-    }
-    if (this.workingDirectory != null && workingDirectory != this.workingDirectory) {
-      return false;
+  void _matches(List<String> command, String workingDirectory, Map<String, String> environment) {
```

Comment:
Using the built in matches gives much better error messages:

Now:
```
00:03 +1 -3: IOSDevice.startApp fails in debug mode when mDNS fails and when Observatory URI is malformed [E]
  Expected: [
              'ios-deploy',
              '--id',
              '123',
              '--bundle',
              '/',
              '--no-wifi',
              '--justlaunch',
              '--args',
              '--enable-dart-profiling --enable-service-port-fallback --disable-service-auth-codes --observatory-port=60700'
            ]
    Actual: [
              'ios-deploy',
              '--id',
              '123',
              '--bundle',
              '/',
              '--no-wifi',
              '--justlaunch',
              '--args',
              '--enable-dart-profiling --enable-service-port-fallback --disable-service-auth-codes --observatory-port=60700 --enable-checked-mode --verify-entry-points'
            ]
     Which: was '--enable-dart-profiling --enable-service-port-fallback --disable-service-auth-codes --observatory-port=60700 --enable-checked-mode --verify-entry-points' instead of '--enable-dart-profiling --enable-service-port-fallback --disable-service-auth-codes --observatory-port=60700' at location [8]
  
```
Before:

```
00:03 +1 -3: IOSDevice.startApp fails in debug mode when mDNS fails and when Observatory URI is malformed [E]
  Expected: true
    Actual: <false>
  ProcessManager was told to execute [ios-deploy, --id, 123, --bundle, /, --no-wifi, --justlaunch, --args, --enable-dart-profiling --enable-service-port-fallback --disable-service-auth-codes --observatory-port=60700 --enable-checked-mode --verify-entry-points] (in null, with environment {PATH: /usr/bin:null, DYLD_LIBRARY_PATH: /path/to/libraries}) but the next process that was expected was [ios-deploy, --id, 123, --bundle, /, --no-wifi, --justlaunch, --args, --enable-dart-profiling --enable-service-port-fallback --disable-service-auth-codes --observatory-port=60700] (in null, with environment {PATH: /usr/bin:null, DYLD_LIBRARY_PATH: /path/to/libraries})}.
```

Reply:
nice